utils/logs.js

```diff
@@ -0,0 +1,122 @@
+const admin = require("firebase-admin");
+const { logType } = require("../constants/logs");
+const usersService = require("../services/dataAccessLayer");
+const { EXTENSION_REQUEST_STATUS } = require("../constants/extensionRequests");
+
+async function getUsersListFromLogs(allLogs) {
+  const userIds = [];
+  for (const log of allLogs) {
+    if (!userIds.includes(log.meta.userId || log.meta.createdBy)) {
+      userIds.push(log.meta.userId || log.meta.createdBy);
+    }
+  }
+  return await usersService.fetchUsersForKeyValues(admin.firestore.FieldPath.documentId(), userIds);
+}
```

Comment:
`getUsersListFromLogs` function iterates over all logs to extract user IDs. Depending on the size of the data, this could be resource-intensive. consider optimizing this process, perhaps by using a set to collect unique user IDs.